File: app/src/main/java/com/projectfox/foxoff/brain/FoxBrainScore.kt
package com.projectfox.foxoff.brain

/**
 * Result of a single analysis cycle.
 */
data class FoxBrainScore(
    /** Probability of sleep from 0.0 to 1.0 */
    val sleepProbability: Float,
    /** Confidence in this score from 0.0 to 1.0 */
    val confidence: Float,
    /** Human-readable reason for the score */
    val reason: String
)

File: ios/FoxCoreKit/Sources/FoxCoreKit/FoxBrainScore.swift
/// Portage direct de `com.projectfox.foxoff.brain.FoxBrainScore` (Kotlin).
/// Résultat d'un cycle d'analyse.
public struct FoxBrainScore: Equatable {
    /// Probabilité de sommeil, de 0.0 à 1.0.
    public let sleepProbability: Float
    /// Confiance dans ce score, de 0.0 à 1.0.
    public let confidence: Float
    /// Raison lisible du score.
    public let reason: String

    public init(sleepProbability: Float, confidence: Float, reason: String) {
        self.sleepProbability = sleepProbability
        self.confidence = confidence
        self.reason = reason
    }
}
